src/hotspot/share/gc/g1/g1CardTable.cpp

```diff
@@ -52,8 +52,8 @@ void G1CardTable::initialize(G1RegionToSpaceMapper* mapper) {
 
   _byte_map_size = mapper->reserved().byte_size();
 
-  _guard_index = cards_required(_whole_heap.word_size()) - 1;
-  _last_valid_index = _guard_index - 1;
+  size_t num_cards = cards_required(_whole_heap.word_size());
+  _last_valid_index = num_cards - 1;
```

Comment:
It looks like `_last_valid_index` can be removed: it's only use is in asserts or very infrequently used methods afaict which imho does not warrant an explicit member now.

Reply:
I agree, but this should probably be dealt with in another PR -- this change doesn't affect the value of `_last_valid_index`.